mautic/mautic
# Sugarcrm integration fullsync

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y 
| Related user documentation PR URL |  https://github.com/mautic/documentation/pull/164
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/3389 and https://github.com/mautic/mautic/issues/3517
| BC breaks? | N
| Deprecations? | N

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Update the SugarCRM plugin to enable full synchronization of Mautic Leads to SugarCRM Contacts and Leads and from Mautic Companies to SugarCRM Accounts, Most functionnalities are similar to Salesforce plugin.
Developped by @canal-web and @webmecanik.

[//]: # ( As applicable: )

#### Steps to test this PR:
1. Create Oauth keys in a SugarCRM 6.X instance https://support.sugarcrm.com/Documentation/Sugar_Developer/Sugar_Developer_Guide_6.5/Application_Framework/Authentication/Oauth/ (Just step 1)
2. On the Mautic side, configure the SugarCRM plugin (Sugar URL, keys, login/password)
3. On the Mautic side, configure the plugin functionnalities
4. On the Mautic side, configure the fields mapping for contacts/leads and companies
5. Run php app/console mautic:integration:fetchleads --integration=Sugarcrm
or php app/console mautic:integration:fetchleads --fetch-all --integration=Sugarcrm to synchronize all leads.
6. Optionnaly install, on the SugarCRM side, the webActivity plugin, and launch
php app/console mautic:integration:pushleadactivity
The zip from the webActivity plugin should be download from :
https://drive.google.com/file/d/0B7f6B2OD0DBrRE0tUmVTcFZNT1k/view
